wandb/sdk/launch/_project_spec.py

```diff
@@ -145,43 +145,68 @@ def get_entry_point(self, entry_point: str) -> "EntryPoint":
             return self._entry_points[entry_point]
         return self.add_entry_point(entry_point)
 
-    def _fetch_project_local(self, api: Api) -> None:
+    def _check_and_download_code_artifacts(self, internal_api: Api) -> bool:
+        public_api = wandb.PublicApi(
+            overrides={"base_url": internal_api.settings("base_url")}
+        )
+
+        run_entity, run_project, run_name = utils.parse_wandb_uri(self.uri)
```

Comment:
nit but might be better to restructure so that this info is already stored on the object? kinda weird to have to run this parse fn over and over but i don't remember where specs are built so maybe there's a good reason for it